pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,343 @@
+	// and not deleted by the user, remove the instance from the queue.
+	if instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		return reconcile.Result{}, nil
+	}
+
+	log.Infof("Reconciling CnsFileAccessConfig with instance: %q from namespace: %q. timeout %q seconds", instance.Name, instance.Namespace, timeout)
+	if !instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Get the virtualmachine instance
+		vm, err := getVirtualMachine(ctx, instance.Spec.VMName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", instance.Spec.VMName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found virtualMachine instance for VM with name: %q: +%v", instance.Spec.VMName, vm)
+
+		// Set ownerRef on CnsFileAccessConfig instance (in-memory) to VM instance
+		setInstanceOwnerRef(instance, instance.Spec.VMName, vm.UID)
+
+		cnsFinalizerExists := false
+		// Check if finalizer already exists.
+		for _, finalizer := range instance.Finalizers {
+			if finalizer == cnsoperatortypes.CNSFinalizer {
+				cnsFinalizerExists = true
+				break
+			}
+		}
+		// Update finalizer and ownerRef on CnsFileAccessConfig instance
+		if !cnsFinalizerExists {
+			// Add finalizer.
+			instance.Finalizers = append(instance.Finalizers, cnsoperatortypes.CNSFinalizer)
+			err = updateCnsFileAccessConfig(ctx, r.client, instance)
+			if err != nil {
+				msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+					instance.Name, instance.Namespace, err)
+				recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+		}
+
+		volumeID, err := util.GetVolumeID(ctx, r.client, instance.Spec.PvcName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get volumeID from pvcName: %q. Error: %+v", instance.Spec.PvcName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Query volume
+		log.Infof("Querying volume: %s for CnsFileAccessConfig request with name: %q on namespace: %q",
+			volumeID, instance.Name, instance.Namespace)
+		volume, err := common.QueryVolumeByID(ctx, r.volumeManager, volumeID)
+		if err != nil {
+			if err.Error() == common.ErrNotFound.Error() {
+				msg := fmt.Sprintf("CNS Volume: %s not found", volumeID)
+				log.Error(msg)
+				setInstanceError(ctx, r, instance, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+			log.Errorf("Failed to query CNS volume: %s with error: %+v", volumeID, err)
+			setInstanceError(ctx, r, instance, "Unable to find the volume in CNS")
```

Comment:
Can you add the error returned by CNS to the instance as well?

Reply:
Done

Reply:
Shouldn't this be a different error message? This  looks pretty much like the one when CNS volume is not found. May be you can set the instance error as "Failed to query CNS volume".